src/nlp/builder.py

```diff
@@ -556,6 +556,10 @@ class GeneratorBasedBuilder(DatasetBuilder):
     (`_split_generators`). See the method docstrings for details.
     """
 
+    def __init__(self, *args, **kwargs):
+        super(GeneratorBasedBuilder, self).__init__(*args, **kwargs)
+        self._writer_batch_size = kwargs.get("writer_batch_size")
```

Comment:
maybe it's nice to use `DEFAULT_MAX_BATCH_SIZE` here as a default for the `.get()` function,  but both ways are fine with me!

Reply:
I think this parameter is more linked to the implementation of the `ArrowWriter`, that's why I think it's better if the default value is set in the writer itself

Reply:
Fair!